tests/Umbraco.Tests.Integration/Testing/UmbracoIntegrationTest.cs

```diff
@@ -189,25 +189,48 @@ protected void ConfigureServices(IServiceCollection services)
 
     private void ExecuteBuilderAttributes(IUmbracoBuilder builder)
     {
-        // todo better errors
+        Type? testClassType = GetTestClassType()
+            ?? throw new Exception($"Could not find test class for {TestContext.CurrentContext.Test.FullName} in order to execute builder attributes.");
 
-        // execute builder attributes defined on method
-        foreach (ConfigureBuilderAttribute builderAttribute in Type.GetType(TestContext.CurrentContext.Test.ClassName)
-                     .GetMethods().First(m => m.Name == TestContext.CurrentContext.Test.MethodName)
-                     .GetCustomAttributes(typeof(ConfigureBuilderAttribute), true))
+        // Execute builder attributes defined on method.
+        foreach (ConfigureBuilderAttribute builderAttribute in GetConfigureBuilderAttributes<ConfigureBuilderAttribute>(testClassType))
         {
             builderAttribute.Execute(builder);
         }
 
-        // execute builder attributes defined on method with param value passtrough from testcase
-        foreach (ConfigureBuilderTestCaseAttribute builderAttribute in Type.GetType(TestContext.CurrentContext.Test.ClassName)
-                     .GetMethods().First(m => m.Name == TestContext.CurrentContext.Test.MethodName)
-                     .GetCustomAttributes(typeof(ConfigureBuilderTestCaseAttribute), true))
+        // Execute builder attributes defined on method with param value pass through from test case.
+        foreach (ConfigureBuilderTestCaseAttribute builderAttribute in GetConfigureBuilderAttributes<ConfigureBuilderTestCaseAttribute>(testClassType))
         {
             builderAttribute.Execute(builder);
         }
     }
 
+    private static Type GetTestClassType()
+    {
+        string testClassName = TestContext.CurrentContext.Test.ClassName;
+
+        // Try resolving the type name directly (which will work for tests in this assembly).
+        Type testClass = Type.GetType(testClassName);
+        if (testClass is not null)
+        {
+            return testClass;
+        }
+
+        // Try scanning the loaded assemblies to see if we can find the class by full name. This will be necessary
+        // for integration test projects using the base classess provided by Umbraco.
+        var assemblies = AppDomain.CurrentDomain.GetAssemblies();
```

Comment:
Was a bit concerned this might be slow, but it won't affect our integration tests as the method will have returned before we hit here.  So it would only be a concern for external integration tests used in packages or projects.

Benchmarking shows this taking 84ms for an integration test that doesn't do anything which takes 917ms.  So < 10% of the time, and will be much less percentage wise for a real test that's actually querying or writing to the database that start from around 2s in our test suite and on my machine.